lib/s3routes/routesUtils.ts

```diff
@@ -1179,12 +1179,19 @@ export function isValidObjectKey(objectKey: string, prefixBlacklist: string[]) {
     if (invalidPrefix) {
         return { isValid: false, invalidPrefix };
     }
-    if (Buffer.byteLength(objectKey, 'utf8') > objectKeyByteLimit) {
-        return { isValid: false };
-    }
     return { isValid: true };
 }
 
+// To be validated only for key creation, such as PutObject, CopyObject, MPU
+export function validateObjectKeyLength(objectKey: string, byteLength?: number) {
```

Comment:
Just a renaming suggestion

```suggestion
export function validateObjectKeyLength(objectKey: string, maxByteLength?: number) {
```